src/actors/symbolication.rs

```diff
@@ -1495,56 +1508,42 @@ impl SymbolicationActor {
             };
 
             Ok((request, minidump_state))
-        });
+        };
 
-        let future = self
+        let result = self
             .threadpool
-            .spawn_handle(lazy.sentry_hub_current().compat())
-            .boxed_local()
-            .compat()
-            .map_err(|_| SymbolicationError::Canceled)
-            .flatten()
-            .then(move |x| {
-                // keep the results until symbolication has finished to ensure we don't drop
-                // temporary files prematurely.
-                drop(cfi_results);
-                x
-            });
+            .spawn_handle(lazy.bind_hub(sentry::Hub::current()))
+            .await
+            .map_err(|_| SymbolicationError::Canceled)?;
 
-        Box::new(future)
+        // keep the results until symbolication has finished to ensure we don't drop
+        // temporary files prematurely.
+        drop(cfi_results);
+        result
     }
 
-    fn do_stackwalk_minidump(
+    async fn do_stackwalk_minidump(
         self,
         scope: Scope,
         minidump: Bytes,
         sources: Vec<SourceConfig>,
-    ) -> impl futures::Future<Output = Result<(SymbolicateStacktraces, MinidumpState), SymbolicationError>>
-    {
-        future_metrics!(
-            "minidump_stackwalk",
-            Some((Duration::from_secs(1200), SymbolicationError::Timeout)),
-            async move {
-                let sources = Arc::new(sources);
-
-                let referenced_modules = self
-                    .get_referenced_modules_from_minidump(minidump.clone())
-                    .compat()
-                    .await?;
+    ) -> Result<(SymbolicateStacktraces, MinidumpState), SymbolicationError> {
+        let future = async move {
+            let sources = Arc::new(sources);
 
-                let cfi_caches = self
-                    .load_cfi_caches(scope.clone(), referenced_modules, sources.clone())
-                    .compat()
-                    .await?;
+            let referenced_modules = self
+                .get_referenced_modules_from_minidump(minidump.clone())
```

Comment:
The only reason we still need these clones is because of the Threadpool requirements internally. I would love to explore if we can get rid of this requirement in a follow-up.

Probably, the proper solution is to get rid of threadpools completely and embrace full asynchronous execution.

Reply:
I would like to hear @flub opinion on this. I’m +1 on just using async directly and have the runtime manage things instead of moving things manually to a threadpool. But I’m not deep enough to understand all the tradeoffs here.

Reply:
huh, while looking at this i (re)realised that we write the entire minidump back to disk just for fun when spawning to procpools.  anyway.

There's is not much gained from `get_referenced_modules_from_minidump` spawning its work onto the symbolication actor's threadpool.  It does have to do a lot of disk IO thanks to the previous paragraph though.  

However in principle we didn't decide to abandon multiple threadpools yet, services will also each control what executor/threadpool they want to run their work on.  Potentially crossing an executor boundary for each service called.  That's perhaps not the best design long-term.  Maybe once we're on a modern tokio and can correctly do disk I/O async the need for this will go away further.